crates/wasm-smith/src/core.rs

```diff
@@ -233,6 +233,68 @@ impl<'a> Arbitrary<'a> for MaybeInvalidModule {
     }
 }
 
+pub struct SingleFunctionModule {
+    signature: FuncType,
+    module: Module,
+}
+
+impl SingleFunctionModule {
+    /// Encode this Wasm module into bytes.
+    pub fn to_bytes(&self) -> Vec<u8> {
+        self.module.to_bytes()
+    }
+
+    /// Build a Wasm module containing a single function with the following
+    /// characteristics:
+    /// - exports the function as `"generated_function"`
+    /// - the function matches the parameters and results of the given
+    ///   `signature` function type
+    /// - the module may contain other Wasm objects (e.g., memories, tables) and
+    ///   these may or may not be exported
+    pub fn build(&mut self, u: &mut Unstructured, allow_invalid: bool) -> Result<()> {
+        let ty = Type::Func(self.signature.into());
+        self.module.valtypes = configured_valtypes(&*self.module.config);
+
+        // Record the type of the single function. Since the module should only
+        // have one function, we do not import additional functions (i.e.,
+        // `arbitrary_imports()`) or other function types (i.e.,
+        // `arbitrary_types()`).
+        self.module.types.push(ty);
+        let ty_idx = self.module.types.len().try_into().unwrap();
```

Comment:
Obviously this kind of thing messes with the meaning of `Config::min_funcs` and `Config::max_funcs`. This `SingleFunctionModule` could grow its own `Config` object but it sure would be nice to figure out a way to re-use the existing `Config`.